scylla-cdc/src/stream_reader.rs

```diff
@@ -63,7 +78,34 @@ impl StreamReader {
         let mut window_begin = self.lower_timestamp;
         let window_size = chrono::Duration::from_std(self.window_size)?;
         let safety_interval = chrono::Duration::from_std(self.safety_interval)?;
+        let mut checkpoint = Checkpoint {
+            timestamp: window_begin.to_std()?,
+            stream_id: self.stream_id_vec[0].clone(),
```

Comment:
If the stream reader is changed to fetch progress from multiple streams, then you will store progress of all those streams in a single checkpoint named by the first stream. While this will work, I dont like it because if we change how the streams are grouped then the application won't correctly restore progress after upgrading the library.

I think we should have a separate checkpoint for each stream. Maybe the task in `start_saving_checkpoint` could keep a list of stream IDs and update several checkpoints instead of only one? As far as restoring progress is concerned, we can start with an easy route: read checkpoints of all streams from the group and start reading all streams from the earliest checkpoint.